bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConnectionPool.java

```diff
@@ -60,6 +60,13 @@ protected synchronized Connection getConnection(Configuration config, String key
 
   @VisibleForTesting
   protected Connection createConnection(Configuration config) throws IOException {
-    return new BigtableConnection(config);
+    return new BigtableConnection(config) {
+      @Override
+      public void close() throws IOException {
+        // Users should not actually close the shared connection. Make sure that if a user does call
```

Comment:
Just a thought: Add a WARN logging to let the user know that he is doing something wrong